### PR TITLE
[[ Bug 20705 ]] Add 'the architecture' syntax to LCB

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

File: libscript/src/system.lcb [hunk 1]
```diff
@@ -56,6 +56,34 @@ begin
...
+    end if
+
+Description:
+Returns a string describing the instruction set architecture
+that is being used by the machine LiveCode is running on.
+The possible values are:
+
+* "x86" - 32-bit Windows, Mac and Linux
+* "x86_64" - 64-bit Windows, Mac and Linux
+* "i386" - iOS simulator
```
Comment (montegoulding) [inline libscript/src/system.lcb]: Hmm.. we don't actually support the 32 bit simulator any more do we? Why wouldn't it return `x86` anyway?`x86_64` should note the iOS 64 bit simulator I think.

Comment (runrevmark) [reply libscript/src/system.lcb]: Yes - we should choose either i386 or x86... And should it be 'x86-64' rather than 'x64_64'?

Comment (montegoulding) [reply libscript/src/system.lcb]: I think the LCS form uses underscore

Comment (livecodeali) [reply libscript/src/system.lcb]: It does, but we could change it in LCB

Comment (montegoulding) [reply libscript/src/system.lcb]: I would keep it as close as possible to the arch portion of the platform ID spec if we can. I'm not sure if we have the info required for arm variants though.

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 64f05bf

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 64f05bf71098eacd218fc7dcb5859e019e200b3b

**Comment by livecode-vulcan** [comment]
😞 test failure 64f05bf71098eacd218fc7dcb5859e019e200b3b

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1062)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1078)
* try-community-js-emscripten-sdk1.35: [failure](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1052)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/5)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/5)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1057)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1079)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1062)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1078)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1061)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1071)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1094)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1057)

File: libscript/src/system.lcb [hunk 2]
```diff
@@ -56,6 +56,34 @@ begin
...
+        -- Do something x86_64 specific
+    end if
+
+Description:
+Returns a string describing the instruction set architecture
+that is being used by the machine LiveCode is running on.
+The possible values are:
+
+* "x86" - 32-bit Windows, Mac and Linux, and iOS simulator
+* "x86_64" - 64-bit Windows, Mac and Linux
```
Comment (montegoulding) [inline libscript/src/system.lcb]: @livecodeali this should note iOS simulator too. I suspect this list could easily get out of date on linux and android without us noticing. Perhaps this should just be something like:

```
* "x86" - 32-bit x86 builds
* "x86_64" - 64-bit x86 builds
* "arm" - 32 bit arm builds
* "arm64" - 64-bit arm builds
* "" - Emscripten
```

File: tests/lcb/stdlib/system.lcb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+module com.livecode.system.tests
+
+use com.livecode.system
+use com.livecode.__INTERNAL._testlib
+
+constant kValidArchs is ["x86", "x86_64", "i386", "arm", "arm64"]
```
Comment (montegoulding) [inline tests/lcb/stdlib/system.lcb]: @livecodeali `i386` is no longer valid and `` is.

File: libscript/src/system.lcb (hunk 2 above)
Comment (montegoulding) [reply libscript/src/system.lcb]: should emscripten be `js` the same as we use for platform id?

Comment (livecodeali) [reply libscript/src/system.lcb]: oh yes, probably!

File: tests/lcb/stdlib/system.lcb (hunk 3 above)
Comment (livecodeali) [reply tests/lcb/stdlib/system.lcb]: thanks

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok 38b024f

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok 38b024f37996587d12fc980efc60da7c012c72f4

**Comment by livecode-vulcan** [comment]
😞 test failure 38b024f37996587d12fc980efc60da7c012c72f4

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1084)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1100)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1074)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/27)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/27)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1079)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1101)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1084)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1100)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1083)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1093)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1116)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1079)

**Comment by montegoulding** [comment]
@livecode-vulcan review ok d75e221

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok d75e22142b8fadfac441d968b2fca05c321748be

**Comment by livecode-vulcan** [comment]
😎 test success d75e22142b8fadfac441d968b2fca05c321748be

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1086)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1102)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1076)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/29)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/29)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1081)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1103)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1086)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1102)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1085)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1095)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1118)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1081)

**Pull request merged**